### PR TITLE
Add Euclidean Distance Transform op

### PR DESCRIPTION
**Pull request by Squadrick**
Fixes [this issue](https://github.com/tensorflow/tensorflow/issues/24410).

Replacement for [this PR](https://github.com/tensorflow/addons/pull/80).

### CODE REVIEW

**Comment by facaiy** [comment]
@sofienbouaziz Hi, Sofien, could you take a look again? Apologies that we rewrote #80 by mistake and have to recreate a new PR.

**Comment by Squadrick** [comment]
@seanpmorgan I fixed the bazel build file format, it should pass the tests now.

**Comment by Squadrick** [comment]
@facaiy @seanpmorgan Anything stopping this from being merged?

File: tensorflow_addons/image/distance_transform_test.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            with self.assertRaisesRegex(
+                    ValueError, "`images` should have rank between 3 and 4"):
+                _ = distance_tranform_ops.euclidean_dist_transform(image)
+
+        image = tf.zeros([2, 4, 3], tf.uint8)
+        with self.assertRaisesRegex(ValueError,
+                                    "`images` must have only one channel"):
+            _ = distance_tranform_ops.euclidean_dist_transform(image)
+
+    def test_all_zeroes(self):
```
Comment (sofienbouaziz) [inline tensorflow_addons/image/distance_transform_test.py]: test_all_zeros?

Comment (sofienbouaziz) [reply tensorflow_addons/image/distance_transform_test.py]: Also sorry for the late review.

Comment (Squadrick) [reply tensorflow_addons/image/distance_transform_test.py]: Almost missed that, thanks.

**Comment by Squadrick** [comment]
@seanpmorgan Fixed the merge conflicts.

**Pull request merged**